File: frontend/desktop/lib/models/requests/update_user_request.dart
class UpdateUserRequest {
  final String firstName;
  final String lastName;
  final String username;
  final String? phoneNumber;

  const UpdateUserRequest({
    required this.firstName,
    required this.lastName,
    required this.username,
    this.phoneNumber,
  });

  Map<String, dynamic> toJson() => {
        'firstName': firstName,
        'lastName': lastName,
        'username': username,
        'phoneNumber': phoneNumber,
      };
}

File: backend/eTicketing.Model/Requests/UpdateUserRequest.cs
using System.ComponentModel.DataAnnotations;

namespace eTicketing.Model.Requests;

public class UpdateUserRequest
{
    [Required(ErrorMessage = "Ime je obavezno")]
    [StringLength(100, ErrorMessage = "Ime ne smije biti duže od 100 karaktera")]
    public string FirstName { get; set; } = string.Empty;

    [Required(ErrorMessage = "Prezime je obavezno")]
    [StringLength(100, ErrorMessage = "Prezime ne smije biti duže od 100 karaktera")]
    public string LastName { get; set; } = string.Empty;

    [Required(ErrorMessage = "Korisničko ime je obavezno")]
    [StringLength(50, MinimumLength = 3, ErrorMessage = "Korisničko ime mora imati između 3 i 50 karaktera")]
    public string Username { get; set; } = string.Empty;

    [StringLength(20, ErrorMessage = "Broj telefona ne smije biti duži od 20 karaktera")]
    public string? PhoneNumber { get; set; }
}
